### ISSUE TITLE
Initial config issues

### ISSUE THREAD

**Issue by vertigo220** [gileli121/WindowTop#399]
When hovering over the toggles and previews for various options during initial config to see what those options do, the popup tooltips only show for a few seconds then hide, not giving nearly enough time to read them. They should stay open as long as the cursor is in position to activate them. They should probably also show faster, as it took some time looking around trying to figure out what some of the options did before I realized they were there.

Also, the tooltip for the "Enable mouse-interact by default" option doesn't make sense. It says "When enabled, you can interact with the window as if it is without having to un-pip the window." I'm not sure, since I don't know what this option does, but I'm assuming the word "if" shouldn't be there.

Another issue is that the "Hide the original window when applying cropping" option and the "Show original window" option appear to do the same thing. The "Show preview on mouse hover," "Enable read mode," and "Enable Live Preview..." options also all seem very similar, and I wonder if it would be better to have the read mode and live preview as sub options for the first one to make it simpler and less confusing.

Maybe there's already an option somewhere for this, but I think there should be a way to exclude videos and/or specific apps from previews. For example, I like the idea of previewing a browser, note app, etc, but I don't want a video playing in the corner to be enlarged every time my mouse goes over it.

When clicking "Other configurations" at the bottom (which is confusing enough because I don't know whether to do that, branching me away from the initial setup, which I'll have to return to and finish afterward, or if I should continue with the initial setup which may include that stuff or maybe should be done first, and it seems as I continue that would likely be the better way), it opens the "Advanced Features Settings" windows. Unfortunately, and ironically considering the main purpose of this app is to control which window is on top, it places this newly opened window _behind_ the current window ("WindowTop Assistant/Wizard"), leading to confusion as it looks like that is the window that just opened (as it is default, and therefore expected, that the newly opened window will be on top), but it's the same window you've been in. It's easy to miss the advanced settings window that in fact just opened.

When advancing to the next page, where you are on the current page affects where you will be on the next page. For example, if you scroll down some then click the "Next" button, you will be scrolled down already on the next page, even at the very bottom, whereas you should always start at the top.

**Comment by gileli121** [comment]
This is a large amount of feedback and suggestions.  
I will respond to this feedback another time.  
Thank you very much for your input and suggestions.

**Comment by gileli121** [comment]
@vertigo220 

> When hovering over the toggles and previews for various options during initial config to see what those options do, the popup tooltips only show for a few seconds then hide, not giving nearly enough time to read them

> They should probably also show faster, as it took some time looking around trying to figure out what some of the options did before I realized they were there.

I see, I noticed that too.  
I think adding a "?" icon to open the tooltip could make it clearer, right? On the other hand, it might make the UI look overly complicated, so I'm not sure. Also, the tooltip definitely shouldn't disappear too quickly and show faster. Maybe if it appears faster, that would be enough to ensure the user doesn’t miss it (instead of adding a "?" icon).


> Also, the tooltip for the "Enable mouse-interact by default" option doesn't make sense. It says "When enabled, you can interact with the window as if it is without having to un-pip the window." I'm not sure, since I don't know what this option does, but I'm assuming the word "if" shouldn't be there.

Mouse interact is this button:
![image](https://github.com/user-attachments/assets/bd23dc09-2e48-4392-8220-dec9eda79604)
This button is available via PiP mode only.

If you disable this in the settings, enabling PiP mode for a window will not activate this feature by default, and you'll need to click this button to enable it. To ensure it is always enabled after activating PiP mode, turn it on in the settings.  

What it does: When enabled, you can interact directly with the window through the small cloned window (referred to as the PiP view). Note that this feature may not work stably with every app, but it works perfectly with Chromium-based browsers, Firefox, Spotify, and more.  

You can see it in action in this video: [YouTube Video (starts at 4:07)](https://youtu.be/V9DOaIFOxes?si=8rYSxs8gEbLwqZj4&t=247)  
As demonstrated in the video, once enabled, you can control the YouTube video directly from the PiP view because it forwards all mouse inputs from the PiP view to the actual window.  

If you have a better way to describe this feature for the tooltip, please let me know!


>Another issue is that the "Hide the original window when applying cropping" option and the "Show original window" option appear to do the same thing. The "Show preview on mouse hover," "Enable read mode," and "Enable Live Preview..." options also all seem very similar, and I wonder if it would be better to have the read mode and live preview as sub options for the first one to make it simpler and less confusing.

First, let me explain what crop mode is: it's essentially PiP mode combined with crop functionality. The crop mode is based on the PiP feature, and you can achieve the same result manually by following these steps:  

1. **Activate PiP Mode**:  
   ![Step 1](https://github.com/user-attachments/assets/8c36d0cf-4d79-41d0-9fb9-7cb81b19fa12)  

2. **Select a Crop Region Inside the PiP View**:  
   ![Step 2](https://github.com/user-attachments/assets/2be15390-cd51-429b-97d0-a1d1ec0e1723)  
   ![Step 3](https://github.com/user-attachments/assets/dd929637-8df1-4856-af44-a980f1bf3eca)  

3. **Resize the PiP View to Match the Size of the Element in the Original Window** (this is the most challenging step).  

At the end, you’ll have a PiP view displaying the selected region of the window. Additionally, you can configure the settings to show the original window from here:  
![Show Original Window](https://github.com/user-attachments/assets/6e32614e-cc42-43a1-9f07-74f5d765a56c)  

In summary, the steps are:  
**PiP → Select Region in PiP View → Resize to Native Size → (Optional) Show/Hide Original Window.**  

### The "Hide the Original Window When Applying Cropping" Setting  

This setting simplifies the process with a shortcut:  
![Shortcut Button](https://github.com/user-attachments/assets/6306aead-79f1-4b49-878d-00e5cfda4341)  
Alternatively, you can enable the crop button in the toolbar:  
![Toolbar Crop Button](https://github.com/user-attachments/assets/459a1f96-da1c-43c7-8ba4-ae5db02af8e6)  

When you click this shortcut, WindowTop automates the process for you:  
- You select the crop region directly on the native window instead of the PiP view.  
![Select Region](https://github.com/user-attachments/assets/a69c822b-c1f8-4712-8e60-ccee9418c5d3)  
![image](https://github.com/user-attachments/assets/4dc0d689-d568-4dca-bbda-eb85a56124fd)

This shortcut simplifies the process to **two clicks**: select the crop region → click "Done."  

### Behavior of the Setting  

- **Enabled**: WindowTop hides the original window when you apply crop mode, making it look like the selected region has been "cut out" while the rest of the window becomes invisible.  
- **Disabled**: Only the selected region is copied into a PiP view, while the original window remains visible.  

### Crop Without Entering PiP Mode  

If you apply crop mode without first entering PiP mode and later remove the PiP view, the next time you activate PiP mode, it will automatically enter crop mode. This happens because WindowTop remembers the previous PiP configuration (PiP + crop).  

I understand this behavior might seem confusing, but I hope this explanation clarifies it!


> Maybe there's already an option somewhere for this, but I think there should be a way to exclude videos and/or specific apps from previews. For example, I like the idea of previewing a browser, note app, etc, but I don't want a video playing in the corner to be enlarged every time my mouse goes over it.

When read mode is enabled, the mouse is cloned into a larger preview, allowing you to see much better what you're doing.  
With this feature enabled, you don’t need to enlarge the PiP view to see your actions because WindowTop magnifies the preview and highlights where the mouse is.  

You can quickly enable or disable this feature from here:  
![Enable/Disable Read Mode](https://github.com/user-attachments/assets/b5059a60-fa15-4a08-b439-57675b5a9772)  

### Saving Behavior per App  

To save the desired behavior for a specific app, you should be able to use this button:  
![Save Configuration](https://github.com/user-attachments/assets/091d220e-c8e1-41c9-beac-ecc644167f52)  

However, this feature doesn’t seem to work. Could you open a bug report for it?  
In any case, I plan to improve this functionality while working on [[feature #283](https://github.com/WindowTop/WindowTop-App/issues/283)](https://github.com/WindowTop/WindowTop-App/issues/283), which is part of the [[roadmap](https://github.com/WindowTop/WindowTop-App/issues/389)](https://github.com/WindowTop/WindowTop-App/issues/389).


# About:

> When clicking "Other configurations" at the bottom (which is confusing enough because I don't know whether to do that, branching me away from the initial setup, which I'll have to return to and finish afterward, or if I should continue with the initial setup which may include that stuff or maybe should be done first, and it seems as I continue that would likely be the better way), it opens the "Advanced Features Settings" windows. Unfortunately, and ironically considering the main purpose of this app is to control which window is on top, it places this newly opened window behind the current window ("WindowTop Assistant/Wizard"), leading to confusion as it looks like that is the window that just opened (as it is default, and therefore expected, that the newly opened window will be on top), but it's the same window you've been in. It's easy to miss the advanced settings window that in fact just opened.

> When advancing to the next page, where you are on the current page affects where you will be on the next page. For example, if you scroll down some then click the "Next" button, you will be scrolled down already on the next page, even at the very bottom, whereas you should always start at the top.


Can you open these as a separate issues (one for each)? It's too much to include all this feedback in one place.  
Thanks for the feedback! I just prefer having it as another issue here to ensure it’s not forgotten.

**Comment by vertigo220** [comment]
> I see, I noticed that too.
> I think adding a "?" icon to open the tooltip could make it clearer, right? On the other hand, it might make the UI look overly complicated, so I'm not sure. Also, the tooltip definitely shouldn't disappear too quickly and show faster. Maybe if it appears faster, that would be enough to ensure the user doesn’t miss it (instead of adding a "?" icon).

I'm not sure a "?" would crowd things, but I can see that it might, and I agree it's not needed with the other changes. You could also consider making the text for the options (the text right above the toggle) show the tooltip when hovered, too, as that's a big target that would be much less likely for the user to miss. And speaking of that, the text should be a consistent size, as some of them are much larger than others, which can be confusing (can result in missing options if looking for the next big text or make it look like options with smaller text are sub-options of those with larger text).

It would also be helpful to have these tooltips not just in the initial setup screens but in the main settings as well, over every option.

> Mouse interact

I'm just saying the text of the tooltip doesn't make sense and seems to either have an extra word or be missing a word. This is what it says:

`When enabled, you can interact with the window as if it is without having to un-pip the window`

and here are a couple examples of what maybe you're trying to say:

`When enabled, you can interact with the window as ~~if~~ it is without having to un-pip the window`
`When enabled, you can interact with the window as if it is _full-size_ without having to un-pip the window`

Hard to provide a better way to word it since I'm not sure what exactly you were going for, and it seems like a typo, but the current wording isn't grammatically correct and doesn't make sense.

> Cropping

I'm not sure you understood what I was saying, but I (think) I understand PiP/cropping, and the GIFs in setup process generally did a good job showing things in action. What I was saying is that there are two options that _seem_, based on their names and lack of info, to do essentially the same thing, but opposite. The first option is

`Hide the original window when applying cropping`

which sound like it does exactly what it does, and the GIFs represent that very clearly. However, the third option down from that one is

`Show original window`

which looks like it's the exact opposite. That is, the first option needs to be _enabled_ to hide the original window, and the second option needs to be _disabled_ to hide it. But they _seem_ like duplicate options, only backwards from each other. I think, especially from your explanation above, that the first option is specifically for cropping, i.e. PiP for a _part_ of a window, and the second option is for PiP _without_ cropping, i.e. the _whole_ window. If that's the case, that should be made more clear in the options. They could be separated, so PiP and Crop each have their own distinct sections (still on the same page since they're so closely related), or there could just be more description given, especially for the second option, and each option could be marked very clearly whether it's related to PiP or cropping. Also, I think they should be made consistent, i.e. both on or both off to show/hide the original window. So for example:

`Hide the original window when applying cropping`

and

`Hide the original window when activating PiP without cropping`

and maybe put them together so it's clear they do the same thing just in different circumstances, because it seems they are essentially the same exact option, only one applies in one case and the other in a different case.

> read mode

Again, I think you misunderstood me. I understand what that mode is (though I wasn't able to figure it out the other night); my concern is that I wouldn't want it to magnify every app. I'd want to specify apps to _not_ magnify when the mouse is over them, so it wouldn't magnify my video player for example. And perhaps this is already possible, as it looks like maybe the box could just be unchecked in the toolbar on that app. Unfortunately, I can't figure out how to get that "Interact" button to test it. And looking through settings trying to figure it out, I'm finding myself confused by what some settings do, e.g. "Enable smart drag for mouse-interact" and "Enable title bar," hence my suggestion earlier that tooltips be added to the regular settings as well. And since I don't have that button or the save button, I can't test it, much less submit a bug report. I'll have to play around with it more later when I have time.

**Comment by gileli121** [comment]
@vertigo220 
Thank you for the extensive feedback. I get the impression that, overall, you're finding it a bit challenging to understand the settings, especially in the advanced configurations.  

I already have a development plan to design the settings in a style that is similar to tree view format, as shown in this video:  
https://github.com/WindowTop/WindowTop-App/issues/375#issuecomment-2356667428 (See from time 0:25)

I’m planning to implement this UI change in the advanced settings screen as well.  

Do you think this will significantly improve the user experience? What do you think?

**Comment by vertigo220** [comment]
Hmm. That might make things easier, but it loses the GIFs, which do a good job of explaining the settings. I don't think there's a need to completely revamp the settings in that way, though if that's already the plan I don't necessarily see anything wrong with it either (aside from losing the animations). I personally find that, regardless of layout or overall design, the make-or-break with settings in an app, especially with such a unique app and with more advanced settings, is to have readily accessible, well-written descriptions of the various options, whether in tooltips, simply text under the options, clicking them to open a window with that info, or whatever. I find far too often apps either don't do this, requiring web searches, scouring the app's help and/or website, or trial-and-error to figure things out, or they do it, but sporadically, so some settings are described while others aren't.

**Comment by gileli121** [comment]
@vertigo220 

Hmm… Right, when I designed this display, I didn’t intend for it to replace what’s in the Quick Start settings. I meant for it to be the style in the advanced settings. I also think that the images and visualizations in the Quick Start settings are a good and helpful idea.

I understand your suggestion about tooltips, and I will implement it.
I’d appreciate it if you could list all the settings in the Quick Start settings that are unclear at first glance, along with an alternative explanation of up to 2 lines that you think would be more suitable (of course, assuming you understood the setting and can provide an alternative explanation).

Additionally, I’d greatly appreciate it if you could also provide an expanded explanation to add to the tooltip.

I have made an effort to document most of the options on these pages:
https://windowtop.info/features/

So, this might help you a lot to understand. Feel free to read the explanations to understand better and use them as a reference for providing explanations.

Thank you very much!
